README.rst

```diff
@@ -10,4 +10,174 @@ test base classes and documentation shared by many XBlocks.
 
 To test the utilities, run::
 
-    nosetests --with-coverage --cover-package xblockutils --cover-html
+    python run_tests.py --with-coverage --cover-package=xblockutils
+
+
+StudioEditableXBlockMixin
+-------------------------
+
+.. code:: python
+
+    from xblockutils.studio_editable import StudioEditableXBlockMixin
+
+This mixin will automatically generate a working ``studio_view`` form
+that allows content authors to edit the fields of your XBlock. To use,
+simply add the class to your base class list, and add a new class field
+called ``editable_fields``, set to a tuple of the names of the fields
+you want your user to be able to edit.
+
+.. code:: python
+
+    @XBlock.needs("i18n")
+    class ExampleBlock(StudioEditableXBlockMixin, XBlock):
+        ...
+        mode = String(
+            display_name="Mode",
+            help="Determines the behaviour of this component. Standard is recommended.",
+            default='standard',
+            scope=Scope.content,
+            values=('standard', 'crazy')
+        )
+        editable_fields = ('mode', 'display_name')
+
+That's all you need to do. The mixin will read the optional
+``display_name``, ``help``, ``default``, and ``values`` settings from
+the fields you mention and build the editor form as well as an AJAX save
+handler.
+
+If you want to validate the data, you can override
+``validate_field_data(self, validation, data)`` and/or
+``clean_studio_edits(self, data)`` - see the source code for details.
+
+Supported field types:
+
+* Boolean:
+  ``field_name = Boolean(display_name="Field Name")``
+* Float:
+  ``field_name = Float(display_name="Field Name")`` 
+* Integer:
+  ``field_name = Integer(display_name="Field Name")`` 
+* String:
+  ``field_name = String(display_name="Field Name")`` 
+* String (multiline):
+  ``field_name = String(multiline_editor=True, resettable_editor=False)``
+* String (html):
+  ``field_name = String(multiline_editor='html', resettable_editor=False)``
+* Any of the above will use a dropdown menu if they have a pre-defined
+  list of possible values.
+* List of undordered unique values (i.e. sets) drawn from a small set of
+  possible values:
+  ``field_name = List(list_style='set', list_values_provider=some_method)``
+
+  - The ``List`` declaration must include the property ``list_style='set'`` to
+    indicate that the ``List`` field is being used with set semantics.
+  - The ``List`` declaration must also define a ``list_values_provider`` method
+    which will be called with the block as its only parameter and which must
+    return a list of possible values.
+* Rudimentary support for List, Dict, and any other JSONField-derived field types
+
+  - ``list_field = List(display_name="Normal List", default=[])``
```

Comment:
It's a bit unclear how this differs from the use of "List" in the above example. Perhaps at L77 instead of just saying "List", you can say "ordered List" or someting that distinguishes it from the list/"set" example previously.